force-app/main/default/triggers/AccountTrigger.trigger

```diff
@@ -0,0 +1,41 @@
+trigger AccountTrigger on Account (before insert, after insert ) {
+    if(Trigger.isBefore && Trigger.isInsert){
+        for(Account acc : Trigger.new){
```

Comment:
The logic is sound! I would just suggest moving to a TriggerHandler to keep the work out of the trigger itself 🙂